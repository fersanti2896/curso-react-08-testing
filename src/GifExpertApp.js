import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Dr. House']);

    // const handleAdd = () => {
    //     setCategories([...categories, 'Hunter']);
    //     /* Otra forma */
    //     // setCategories(cat => [...cat, 'Hunter']);
    // }

    return (
        <>
            <h2>GifExpert App</h2>
            <AddCategory setCategories = {setCategories}/>
            <hr />
            <ol>
                {
                    categories.map(category => (
                        <GifGrid 
                            key = {category}
                            category = {category}
                        />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;