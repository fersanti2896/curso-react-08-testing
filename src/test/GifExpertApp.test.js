import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp'

describe('Haciendo pruebas en componente GifExpertApp', () => {
    test('Debe mostrar <GifExpertApp /> correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['Yugioh', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});