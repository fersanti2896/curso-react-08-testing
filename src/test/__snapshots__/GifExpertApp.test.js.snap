// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo pruebas en componente GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Yugioh"
      key="Yugioh"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Haciendo pruebas en componente GifExpertApp Debe mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
